vllm/model_executor/layers/quantization/fp8.py

```diff
@@ -629,6 +653,24 @@ def create_weights(self, layer: Module, num_experts: int, hidden_size: int,
             layer.w13_input_scale = None
             layer.w2_input_scale = None
 
+    def _gaudi_weight_wrapper(self, weight_loader):
+        """Wrapper for Gaudi weight conversion."""
+
+        def wrapper(*args, **kwargs):
+            # args[0] is parameter, args[1] is loaded_weight
+            # weights will be always in fp8, but scales will be in fp32,
+            # so we can detect it by dtype
+            loaded_weight = args[1]
+            if loaded_weight.dtype == torch.float8_e4m3fn:
+                loaded_weight.data = (loaded_weight.data.float() * 0.5).to(
+                    torch.float8_e4m3fn)
+            else:
+                loaded_weight.data = (loaded_weight.data * 2.0)
```

Comment:
Should we treat `weight_scale` and `weight_scale_inv` differently?

Reply:
At this moment, our only use case for this is Deepseek, so we don't need to differentiate that.